libraries/AP_Proximity/AP_Proximity_LD06.cpp

```diff
@@ -0,0 +1,166 @@
+    get_readings();
+
+    // Check if the data is being received correctly and sets Proximity Status
+    if ((_last_distance_received_ms == 0)) {
+        set_status(AP_Proximity::Status::NoData);
+        Debug(1, "LIDAR NO DATA");
+    } else {
+        set_status(AP_Proximity::Status::Good);
+    }
+}
+
+// Stores the maximum distance that the sensor supports
+float AP_Proximity_LD06::distance_max() const
+{
+    return 12.0f;  // LD06 has a maximum range of 12m
+}
+
+// Stores the minimum distance that the sensor supports
+float AP_Proximity_LD06::distance_min() const
+{
+    return 0.02f;  // LD06 has a minimum range of 2 cm (0.02 m)
+}
+
+// Called repeatedly to get the readings at the current instant
+void AP_Proximity_LD06::get_readings()
+{
+    if (_uart == nullptr) {
+        return;
+    }
+
+    // Store the UART input in an array
+    uint32_t nbytes = _uart->available();
+
+    // Loops through all bytes that were received
+    while (nbytes-- > 0) {
+
+        // Gets and logs the current byte being read
+        uint8_t c = _uart->read();
+        Debug(2, "UART READ %x <%c>", c, c); //show HEX values
+
+        // Stores the byte in an array if the byte is a start byte or we have already read a start byte
+        if(c == LD_START_CHAR || _response_data) {
+
+            // Sets to true if a start byte has been read, default false otherwise
+            _response_data = true;
+
+            // Stores the next byte in an array
+            _response[_byte_count] = c;
+            _byte_count++;
+
+            // The response should have 47 bytes in total
+            if (_byte_count == sizeof(_response)) {
+                Debug(2,"LiDAR Data Received");
+
+                parse_response_data();
+                _byte_count = 0
+            }
+        } else {
+            Debug(1, "Unknown LiDAR Data Received")
+        }
+    }
+
+    // Store for debugging purposes
+    _last_request_ms = AP_HAL::millis();
+}
+
+// Parses the data packet received from the LiDAR
+void AP_Proximity_LD06::parse_response_data()
+{ 
+
+    // Data interpretation based on:
+    // http://wiki.inno-maker.com/display/HOMEPAGE/LD06?preview=/6949506/6949511/LDROBOT_LD06_Development%20manual_v1.0_en.pdf
+
+    // Second byte in array stores length of data - not used but stored for debugging
+    uint8_t data_length = _response[1];
+
+    // Respective bits store the radar speed, start/end angles
+    // Use bitwise operations to correctly obtain correct angles
+    // Divide angles by 100 as per manual
+    uint16_t radar_speed = (_response[3] << 0x08) | (_response[2]);
+    uint16_t start_angle = (_response[5] << 0x08) | (_response[4]) / 100;
+    uint16_t end_angle = (_response[43] << 0x08) | (_response[42]) / 100;
+
+    // Indicates where the payload data begins
+    uint8_t start_payload = 6;
+
+    // Used to calculate the angle inside the data packet
+    uint8_t angle_inc_counter = 1;
```

Comment:
Shouldn't this start at 0?  So the first sample's output angle is the start angle?

![image](https://user-images.githubusercontent.com/69225461/155079936-9873ef93-8d34-46ec-8ad9-bdbd40d1eb6b.png)


```suggestion
    uint8_t angle_inc_counter = 0;
```